commit 951a8cdba307986312182656fcc4fb48142f46cc (refs/changes/65/1765/2)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-04-10T16:52:25+02:00 (2 years, 6 months ago)
    
    TOOLS-1741 sdcadm create: Warning about --skip-ha-check should be more scary

diff --git a/lib/cli/do_create.js b/lib/cli/do_create.js
index 6f3e3de..f72b1bf 100644
--- a/lib/cli/do_create.js
+++ b/lib/cli/do_create.js
@@ -137,7 +137,7 @@ Create.prototype.execute = function cExecute(opts, args, cb) {
                 changes: changes,
                 progress: self.progress,
                 uuid: self.uuid,
-                skipHACheck: opts.skip_ha_check,
+                skipHACheck: opts.dev_allow_multiple_instances,
                 forceDataPath: true
             }, function (err, plan_) {
                 plan = plan_;
@@ -217,13 +217,12 @@ Create.prototype.help = (
     'Usage:\n' +
     '       {{name}} create <svc>\n\n' +
     'Note that in order to create an instance of some services the option\n' +
-    '--skip-ha-ready must be specified, given that those services are not\n' +
-    'supposed to have more than one instance. There are also some services\n' +
-    'which are not allowed to have more than one instance, or services\n' +
-    'whose instances should not be created using this tool, like manatee or\n' +
-    'binder. Finally, the first instance of some services should not be\n' +
-    'created using this tool when there is an alternate choice provided by\n' +
-    'post-setup subcommand.\n'
+    '--dev-allow-multiple-instances must be specified, given that those\n' +
+    'services are not supposed to have more than one instance. There are\n' +
+    'also some services whose instances should not be created using\n' +
+    'this tool, like manatee or binder. Finally, the first instance of some\n' +
+    'services should not be created using this tool when there is an\n' +
+    'alternate choice provided by post-setup subcommand.\n'
 );
 
 
@@ -276,10 +275,10 @@ do_create.options = [
         help: 'Answer yes to all confirmations.'
     },
     {
-        names: ['skip-ha-check'],
+        names: ['dev-allow-multiple-instances'],
         type: 'bool',
         help: 'Allow create the instance even if the service is not ' +
-              'HA ready.'
+              'HA ready (for development purposes).'
     }
 ];
 
diff --git a/lib/procedures/index.js b/lib/procedures/index.js
index f81f7d2..d254287 100644
--- a/lib/procedures/index.js
+++ b/lib/procedures/index.js
@@ -629,11 +629,14 @@ function coordinatePlan(opts, cb) {
                         handle.push(change);
                     } else {
                         if (avoid.indexOf(sName) === -1) {
-                            // Let the user know about --skip-ha-check
-                            err = new UsageError(
-                                'Must provide \'--skip-ha-check\' option in ' +
-                                'order to create another instance of ' +
-                                sName);
+                            err = new UsageError(format(
+                                'The "%s" service does not support proper ' +
+                                'operation with multiple instances. Running ' +
+                                'multiple instances is unsupported.\n(For ' +
+                                'development of Triton, this guard can be ' +
+                                'skipped with ' +
+                                '"--dev-allow-multiple-instances")',
+                                sName));
                         } else {
                             remaining.push(change);
                         }
diff --git a/man/man1/sdcadm.1.ronn b/man/man1/sdcadm.1.ronn
index 096cae2..0b419c8 100644
--- a/man/man1/sdcadm.1.ronn
+++ b/man/man1/sdcadm.1.ronn
@@ -260,13 +260,12 @@ rollback the services/instances listed into the provided plan file.
 Create an instance for an existing SDC service.
 
 Note that in order to create an instance of some services the option
-`--skip-ha-ready` must be specified, given that those services are not
-supposed to have more than one instance. There are also some services
-which are not allowed to have more than one instance, or services
-whose instances should not be created using this tool, like manatee or
-zookeeper. Finally, the first instance of some services should not be
-created using this tool when there is an alternate choice provided by
-`post-setup` subcommand.
+--dev-allow-multiple-instances must be specified, given that those
+services are not supposed to have more than one instance. There are
+also some services whose instances should not be created using
+this tool, like manatee or binder. Finally, the first instance of some
+services should not be created using this tool when there is an
+alternate choice provided by post-setup subcommand.
 
 
 `-h, --help`
@@ -284,8 +283,9 @@ created using this tool when there is an alternate choice provided by
 `-y, --yes`
     Answer yes to all confirmations.
 
-`--skip-ha-check`
-    Allow create the instance even if the service is not HA ready.
+`--dev-allow-multiple-instances`
+    Allow create the instance even if the service is not HA ready
+    (for development purposes).
 
 ### sdcadm check-config \[options\]
 
